homeassistant/components/dlna_dmr/media_player.py

```diff
@@ -278,7 +278,7 @@ async def async_ssdp_callback(
             await self._device_disconnect()
 
         if change == ssdp.SsdpChange.ALIVE and not self._device:
-            location = info[ssdp.ATTR_SSDP_LOCATION]
+            location = info.ssdp_location or ""
```

Comment:
Is empty string a valid argument to the below function call?

Reply:
In this case, getting `SsdpChange.ALIVE` implies that `info.ssdp_location` should be a non-empty string. If it isn't, this function should log an error and bail out.

In answer to your question, no it's not a valid argument. But it will eventually be cause a problem in the call to `domain_data.upnp_factory.async_create_device(location)`, which will raise a `UpnpError`, that then gets caught here and logged. So in a long-winded way it'll be handled gracefully.

Reply:
If we agree on a standard/recommended practise, I can implement in #60794